File: typescript/arrays/linearSearch.ts
// linear search 

function linearSearch(arr: number[], key: number): number {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === key) {
            return i;
        }
    }
    return -1;
}

console.log(linearSearch([1, 2, 3, 4, 5], 3));

File: python/02-linearSearch/linear_search.py
def linear_search(arr, target):
    for i in range(0, len(arr)):
        if arr[i] == target:
            return i
    return -1

if __name__ == "__main__":
    arr = [1, 2, 3, 4, 5, 6]
    target = 4
    ans = linear_search(arr, target)
    print("target found at index: ", ans)

